test/runtime_tests.cc

```diff
@@ -56,6 +58,20 @@ namespace detail {
 		}
 	};
 
+	struct runtime_testspy {
```

Comment:
I wonder if testing the **actual** runtime thread names is necessary. Just testing `set_current_thread_name()` and `set_thread_name()` on artificial threads created in the tests should be enough and would avoid declaring all these testspies.

Reply:
While all those test spies aren't great, I feel like checking the actual thread names is valuable, since setting those could feasibly break independently of the utility functions.

Reply:
Sure, I don't feel to strongly about this.